### PR TITLE
Redirect search pages to home when no search query is present

### LINKED ISSUE #596: Redirect /search/ page to the Openverse homepage
## Problem

Currently, when you go to `https://wordpress.org/openverse/search/`, you see an image page that confusingly states that you got no results for " ", and a `Load more` button. The page is not linked from anywhere, so you can only get there if you type in the exact address. 

Screen Shot 2022-01-12 at 6 43 32 PM

## Description

It is better for user experience to redirect this page to the home page (`https://wordpress.org/openverse/`), so that even if the user gets to the `/search/` address, they are not confused with incorrect text and impossible actions.

## Alternatives

We could design a special page for this address, but considering that it is not linked from anywhere, it doesn't seem to be a good use of time and resources.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by zackkrida**
## Fixes

Fixes #596 by @obulat

## Description

Adds an anonymous middleware to the root `/search` route so that any search path without a search query paremeter is redirected to the homepage.

## Testing Instructions

Visit `/search/` and all child routes without a query param and observe homepage redirection. You should not be redirected if there is a search term. 

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/pages/search.vue [hunk 1]
```diff
@@ -53,6 +53,9 @@ export default defineComponent({
     VSearchGrid,
     VSkipToContentContainer,
   },
+  middleware({ route, redirect }) {
+    if (!route.query.q) return redirect('/')
```
Comment (zackkrida) [inline src/pages/search.vue:57]:
```suggestion
    if (!route.query.q || !route.query.creator) return redirect('/')
```

**Review by obulat — APPROVED** [review]
Nice that it was such a simple change :)

File: test/playwright/e2e/redirect-queryless-searches.spec.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+/**
+ * Openverse searches without a search term / query should
+ * be redirected to the Openverse homepage.
+ */
```
Comment (zackkrida) [inline test/playwright/e2e/redirect-queryless-searches.spec.ts:4]:
```suggestion
/**
 * Searches without a search term should
 * redirect to the homepage.
 */
```

**Comment by zackkrida** [comment]
I've added a new e2e test file, @sarayourfriend

**Review by dhruvkb — APPROVED** [review]

**Pull request merged**